### ISSUE TITLE
Repair utility is too slow

### ISSUE THREAD

**Issue by jamieb22** [orientechnologies/orientdb#7030]
## OrientDB Version, operating system, or hardware.
- v2.2.13 

## Operating System
- [ X] Linux

## Expected behavior and actual behavior

Running repair on a corrupted database is very slow. It take 31 hours to complete the repair process large database of 37GB. We are not even sure how the database can be so big. We are not storing alot of data inside it.

**Comment by laa** [comment]
@jamieb22 what do you mean by database repair ? could you provide list of files inside DB directory together with their sizes?

**Comment by jamieb22** [comment]
I afraid its too late for me to get this info, since the database is already repaired. I am talking about http://orientdb.com/docs/2.2/Console-Command-Repair-Database.html. This command takes a very long time, and in the end it didn't really repair the db, we had to drop data to make it work.

**Comment by laa** [comment]
what kind of problem did you have ? Could you describe in details ?

**Comment by jamieb22** [comment]
We received a NullPointerException when accessing a vertex, so we decided a repair was necessary and ran the Orient DB repair command. After waiting more than 31 hours for repair to complete, the repair tool did not repair the database. In the end, we had to drop the data.

Orient DB generated a NullPointerException when connecting to it...

2016-12-20 11:23:22 t.c.s.a.d.DatabaseService [ERROR] failed to connect to database:null

java.lang.NullPointerException: null
               at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserializeValue(ORecordSerializerBinaryV0.java:395)
               at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.readEmbeddedMap(ORecordSerializerBinaryV0.java:565)
               at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserializeValue(ORecordSerializerBinaryV0.java:467)
               at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserialize(ORecordSerializerBinaryV0.java:275)
               at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinary.fromStream(ORecordSerializerBinary.java:80)
               at com.orientechnologies.orient.core.record.impl.ODocument.deserializeFields(ODocument.java:1851)
               at com.orientechnologies.orient.core.record.impl.ODocument.checkForFields(ODocument.java:2626)
               at com.orientechnologies.orient.core.record.impl.ODocument.setDirtyNoChanged(ODocument.java:1468)
               at com.orientechnologies.orient.core.record.impl.OSimpleMultiValueChangeListener.onAfterRecordChanged(OSimpleMultiValueChangeListener.java:58)
               at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OSBTreeRidBag.fireCollectionChangedEvent(OSBTreeRidBag.java:949)
               at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OSBTreeRidBag$RIDBagIterator.remove(OSBTreeRidBag.java:378)
               at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeEdges(OrientBaseGraph.java:1996)
               at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeInverseEdge(OrientBaseGraph.java:2122)
               at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeEdges(OrientBaseGraph.java:2025)
               at com.tinkerpop.blueprints.impls.orient.OrientGraph.removeEdgesInternal(OrientGraph.java:440)
               at com.tinkerpop.blueprints.impls.orient.OrientVertex.remove(OrientVertex.java:508)
               at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeVertex(OrientBaseGraph.java:722)

**Comment by laa** [comment]
@tglman that seems like an issue in serializer, could you look at it?
@jamieb22 about your initial request, we can not tell without statistic of how much resources were spent on each method why it was slow, but as I can understand you do not have initial data to make such statistics. But we try to fix NPE by the stack trace.

**Comment by jamieb22** [comment]
Yes, the data has been dropped. It is very difficult to work with such a large amount of data.

**Comment by laa** [comment]
@lvca why you reassing me  ? that is serialization exception

**Comment by laa** [comment]
```
java.lang.NullPointerException: null
at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.deserializeValue(ORecordSerializerBinaryV0.java:395)
at com.orientechnologies.orient.core.serialization.serializer.record.binary.ORecordSerializerBinaryV0.readEmbeddedMap(ORecordSerializerBinaryV0.java:565)
```

**Comment by tglman** [comment]
hi @jamieb22,

this bug is already fixed in the 2.2.14, here is the commit:https://github.com/orientechnologies/orientdb/commit/21d0f73619dc7a9cb70104a726f8f768854959a7.

by the way this can happen only if you create properties using "UNSAFE", be aware that unsafe need to be used carefully.

Regards
